packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ClientProvider.cs

```diff
@@ -208,20 +200,24 @@ protected override FieldProvider[] BuildFields()
         {
             List<FieldProvider> fields = [EndpointField];
 
-            if (_apiKeyAuthFields != null)
+            // Skip auth fields for sub-clients
+            if (ClientOptions != null)
             {
-                fields.Add(_apiKeyAuthFields.AuthField);
-                fields.Add(_apiKeyAuthFields.AuthorizationHeaderField);
-                if (_apiKeyAuthFields.AuthorizationApiKeyPrefixField != null)
+                if (_apiKeyAuthFields != null)
                 {
-                    fields.Add(_apiKeyAuthFields.AuthorizationApiKeyPrefixField);
+                    fields.Add(_apiKeyAuthFields.AuthField);
+                    fields.Add(_apiKeyAuthFields.AuthorizationHeaderField);
+                    if (_apiKeyAuthFields.AuthorizationApiKeyPrefixField != null)
+                    {
+                        fields.Add(_apiKeyAuthFields.AuthorizationApiKeyPrefixField);
+                    }
                 }
-            }
 
-            if (_oauth2Fields != null)
-            {
-                fields.Add(_oauth2Fields.AuthField);
-                fields.Add(_oauth2Fields.AuthorizationScopesField);
+                if (_oauth2Fields != null)
+                {
+                    fields.Add(_oauth2Fields.AuthField);
+                    fields.Add(_oauth2Fields.AuthorizationScopesField);
+                }
```

Comment:
this is quite complicated - would it be simpler if we assign null to those `XXAuthFields` in the ctor?

Reply:
updated